optimum/intel/ipex/modeling_base.py

```diff
@@ -353,12 +353,12 @@ def _reorder_cache(self, *args, **kwargs):
     def prepare_inputs_for_generation(self, *args, **kwargs):
         return self.model.prepare_inputs_for_generation(*args, **kwargs)
 
-    def _supports_num_logits_to_keep(self) -> bool:
+    def _supports_logits_to_keep(self) -> bool:
         """
-        Return True if the current model supports the keyword argument `num_logits_to_keep` in forward()
+        Return True if the current model supports the keyword argument `logits_to_keep` in forward()
         to save memory. Checking it in this way allows to avoid using a new model attribute.
         """
-        return "num_logits_to_keep" in set(inspect.signature(self.model.forward).parameters.keys())
+        return "logits_to_keep" in set(inspect.signature(self.model.forward).parameters.keys())
```

Comment:
what about something to keep compatibility with transformers version lower than 4.49 
```suggestion
        logits_to_keep_name = "logits_to_keep" if is_transformers_version(">", "4.49") else "num_logits_to_keep"
        return logits_to_keep_name in set(inspect.signature(self.model.forward).parameters.keys())
```